docs/topics/db/transactions.txt

```diff
@@ -290,45 +290,47 @@ Performing actions after commit
 
 Sometimes you need to perform an action related to the current database
 transaction, but only if the transaction successfully commits. Examples might
-include a `Celery`_ task, an email notification, or a cache invalidation.
+include a background task, an email notification, or a cache invalidation.
 
-.. _Celery: https://pypi.org/project/celery/
-
-Django provides the :func:`on_commit` function to register callback functions
-that should be executed after a transaction is successfully committed:
+:func:`on_commit` allows you to register callbacks that will be executed after
+the open transaction is successfully committed:
 
 .. function:: on_commit(func, using=None, robust=False)
 
-Pass any function (that takes no arguments) to :func:`on_commit`::
+Pass a function, or any callable, to :func:`on_commit`::
 
     from django.db import transaction
 
-    def do_something():
-        pass  # send a mail, invalidate a cache, fire off a Celery task, etc.
+    def send_welcome_email():
+        ...
 
-    transaction.on_commit(do_something)
+    transaction.on_commit(send_welcome_email)
 
-You can also bind arguments to your function using :func:`functools.partial`::
+Callbacks will not be passed any arguments, but you can bind them with
+:func:`functools.partial`::
 
     from functools import partial
 
-    transaction.on_commit(partial(some_celery_task.delay, 'arg1'))
+    for user in users:
+        transaction.on_commit(
+            partial(send_invite_email, user=user)
+        )
```

Comment:
more realistic example, and a case where you need `partial()` to avoid the [late-binding bug](https://adamj.eu/tech/2022/08/22/use-partial-with-djangos-transaction-on-commit/)

I also considered an extra warning block about how `partial` avoids late-binding bugs, compared to inner function or lambda, but it’s quite hard to word

Could link to my blog post? Or add a section below? I'm worried that if we don't point out the issue somehow, people will still write late-binding-bugged invocations. Still easy to define a function inside the loop.

Reply:
Fair enough, we can leave it if you think so. I guess it’s the responsibility of tools like flake8-bugbear to detect such problems, since it’s a general python problem with callbacks, not specific to `transaction.on_commit`.